server/datastore/mysql/campaigns.go

```diff
@@ -133,13 +145,27 @@ func (ds *Datastore) CleanupDistributedQueryCampaigns(ctx context.Context, now t
 		fleet.QueryWaiting, now.Add(-1*time.Minute),
 		fleet.QueryRunning, now.Add(-24*time.Hour))
 	if err != nil {
-		return 0, ctxerr.Wrap(ctx, err, "updating distributed query campaign")
+		return 0, nil, ctxerr.Wrap(ctx, err, "updating distributed query campaign")
 	}
 
 	exp, err := result.RowsAffected()
 	if err != nil {
-		return 0, ctxerr.Wrap(ctx, err, "rows affected updating distributed query campaign")
+		return 0, nil, ctxerr.Wrap(ctx, err, "rows affected updating distributed query campaign")
 	}
 
-	return uint(exp), nil
+	const getInactiveStmt = `
+		SELECT
+			id
+		FROM
+			distributed_query_campaigns
+		WHERE
+			status = ? AND
+			created_at > ?
+`
+	// using the writer so we catch the ones we just marked as completed
+	err = sqlx.SelectContext(ctx, ds.writer(ctx), &recentInactive, getInactiveStmt, fleet.QueryComplete, now.AddDate(0, 0, -7))
```

Comment:
We have customer/s [*] that will run **one** live query for each host once a day (~10k hosts). Will this be an issue? (In that scenario this will return up 70k rows, right?)

[*] https://github.com/fleetdm/fleet/issues/16805

Reply:
Hmm... I mean, even then it's still just 546KB (it's not small, but not crazy huge either, I think we deal with much bigger data in memory for e.g. vulnerabilities?).

But my concern with this use is related to how live queries are stored in redis, we did implement it the way they are with the expectation that the number of live queries would be small (https://github.com/fleetdm/fleet/blob/main/server/live_query/redis_live_query.go#L20-L22). I wonder if that should be reconsidered or maybe even configurable depending on the planned use of such queries...

Reply:
(not as part of this PR, of course, that's a bigger undertaking)

Reply:
OK, fair enough.

Any issues with a Redis `SREM` call with ~70k arguments or a `DEL` call with ~140k arguments?

Reply:
That's a good question... I don't think the size/number of args is too problematic in and of itself (I found this on stackoverflow, dated a bit but that's coming from someone from Redis Labs), the 140K keys would be between 2-3MB. The bigger issue, I think, is the fact that this may block the redis server for a significant amount of time - "SREM" is a "fast" operation, but not "DEL", although we could use [unlink](https://redis.io/commands/unlink/) which is a newer fast delete operation (because it is async). I don't remember what's the minimal redis version we officially support, I would guess it's 4 if not 5 so I could change the code to use `UNLINK`.  Even `SREM` could be blocking for a significant time at those numbers, though...

I feel like in general uses, this would be fine, but for those heavy users of live queries, a more important refactoring/overhaul of how live queries get stored could be better? Do we currently run into issues running those 1 LQ/day/host?